### DISCUSSION TITLE
Add Distance to Time Conductor

### DISCUSSION THREAD

**Discussion by milesbjackson** [nasa/openmct#5873] [Community Q&A]
Hello all,

Love Open MCT. I am using it for plotting F1 car data live and it works a peach!

I have data coming in with timestamps, but I am also able to get 'distance around the track' for each data point. With circuit racing doing many laps, we like to overlay the plots according to distance around the track so we can train the driver on where they are gaining or losing over the competition.

I'd like to add a Fixed Distance to the Time Conductor. Is that perhaps the best way? Instead of timestamp it plots it according to distance from the startline?

The other way was having to duplicate all the widgets and make distance based versions of them. But i like the idea of being able to switch views between fixed distance and local clock with the time conductor. Does anyone foresee any large pitfalls if i attempt to add distance based to the time conductor as a menu item and back end functionality?

Thanks in advance,

Miles

**Comment by jvigliotta** [comment]
Hey @milesbjackson ! That's a pretty cool idea. Take a look at our [TimeAPI](https://github.com/nasa/openmct/blob/master/API.md#time-api) . I think if you were to create a new TimeSystem (in this case, it will be based off of distance) and a [clock](https://github.com/nasa/openmct/blob/master/API.md#clocks) that also ticks on distance updates, then you could pull this off. You would probably also need to add a [TimeFormat](https://github.com/nasa/openmct/blob/master/API.md#telemetry-formats) so you can convert distances. It's not a small endeavor, but it's definitely possible.

**Reply by milesbjackson** [reply]
Ah, excellent advice. I will try that and report back! Thanks for pointing me in those directions. This looks doable. Work, but doable!

Miles
